optuna/integration/allennlp.py

```diff
@@ -9,17 +10,142 @@
+    if pruner_class is None:
+        return None
+
+    pruner_params = _get_environment_variables_for_pruner()
+
+    pruner = getattr(optuna.pruners, pruner_class, None)
+
+    if pruner is None:
+        return None
+
+    return pruner(**pruner_params)
+
+
+def _infer_and_cast(value: Optional[str]) -> Any:
+    """Infer and cast a string to desired types.
+
+    We are only able to set strings as environment variables.
+    However, parameters of a pruner could be integer, float,
+    boolean, or else. We infer and cast environment variables
+    to desired types.
+
+    """
+    if value is None:
+        return None
+
+    result = value  # type: Any
+
+    try:
+        result = int(value)
+    except ValueError:
+        try:
+            result = float(value)
+        except ValueError:
+            if result == "True":
+                result = True
+            elif value == "False":
+                result = False
+
+    return result
+
+
+def _get_environment_variables_for_trial() -> Dict[str, Optional[str]]:
+    return {
+        "study_name": os.getenv("{}_OPTUNA_ALLENNLP_STUDY_NAME".format(PPID)),
+        "trial_id": os.getenv("{}_OPTUNA_ALLENNLP_TRIAL_ID".format(PPID)),
+        "storage": os.getenv("{}_OPTUNA_ALLENNLP_STORAGE_NAME".format(PPID)),
+        "monitor": os.getenv("{}_OPTUNA_ALLENNLP_MONITOR".format(PPID)),
+    }
+
+
+def _get_environment_variables_for_pruner() -> Dict[str, Optional[str]]:
+    keys = os.getenv("{}_OPTUNA_ALLENNLP_PRUNER_KEYS".format(PPID))
+    if keys is None:
+        return {}
+
+    kwargs = {}
+    for key in keys.split(","):
+        value = os.getenv("{}_OPTUNA_ALLENNLP_{}".format(PPID, key))
+        kwargs[key] = _infer_and_cast(value)
+
+    return kwargs
+
+
+def _fetch_pruner_config(trial: optuna.Trial) -> Dict[str, Any]:
+    pruner = trial.study.pruner
+    kwargs = {}  # type: Dict[str, Any]
+
+    if isinstance(pruner, optuna.pruners.HyperbandPruner):
+        kwargs["min_resource"] = pruner._min_resource
+        kwargs["max_resource"] = pruner._max_resource
+        kwargs["reduction_factor"] = pruner._reduction_factor
+    elif isinstance(pruner, optuna.pruners.PercentilePruner):
+        kwargs["percentile"] = pruner._percentile
+        kwargs["n_startup_tirals"] = pruner._n_startup_trials
+        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
+        kwargs["interval_steps"] = pruner._interval_steps
+    elif isinstance(pruner, optuna.pruners.ThresholdPruner):
+        kwargs["lower"] = pruner._lower
+        kwargs["upper"] = pruner._upper
+        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
+        kwargs["interval_steps"] = pruner._interval_steps
+    elif isinstance(pruner, optuna.pruners.SuccessiveHalvingPruner):
+        kwargs["min_resource"] = pruner._min_resource
+        kwargs["reduction_factor"] = pruner._reduction_factor
+        kwargs["min_early_stopping_rate"] = pruner._min_early_stopping_rate
+
```

Comment:
I think we'll end up trying to re-instantiate a pruner with empty kwargs `{}` afterwards if an unrecognized pruner is given here. I don't think it's expected and we might want to raise a warning here in an `else`, if not an error.

Reply:
You're right. I added `else` in 0ddc47f.